mne/gui/__init__.py

```diff
@@ -247,6 +247,86 @@ def locate_ieeg(info, trans, aligned_ct, subject=None, subjects_dir=None,
     return gui
 
 
+@verbose
+def view_vol_stc(stcs, freq_first=True, subject=None, subjects_dir=None,
+                 src=None, inst=None, show=True, block=False, verbose=None):
+    """View a source time course estimate.
+
+    Parameters
+    ----------
+    stcs : list of list | generator
+        The source estimates. List of lists or generators for epochs
+        and frequencies (i.e. using
+        :func:`mne.minimum_norm.apply_inverse_tfr_epochs` or
+        :func:`mne.beamformer.apply_dics_tfr_epochs`-- in this case
+        use ``freq_first=False``). Lists of source estimates across
+        frequencies (e.g. :func::func:`mne.beamformer.apply_dics_csd`)
+        and lists of source estimates across epochs
+        (e.g. :func:`mne.minimum_norm.apply_inverse_epochs` and
+        :func:`mne.beamformer.apply_dics_epochs`--in these
+        case use ``freq_first=False``) are also allowed. Single
+        source estimates (e.g. :func:`mne.minimum_norm.apply_inverse`
+        and :func:`mne.beamformer.apply_dics`) are also allowed
+        (``freq_first`` will not be used in this case).
+    freq_first : bool
+        If frequencies are the outer list of ``stcs`` use ``True``.
+    %(subject)s
+    %(subjects_dir)s
+    src : instance of SourceSpaces
+        The volume source space for the ``stc``.
+    inst : EpochsTFR | AverageTFR | None
+        The time-frequency or data object to use to plot topography.
+    show : bool
+        Show the GUI if True.
+    block : bool
+        Whether to halt program execution until the figure is closed.
+    %(verbose)s
+
+    Returns
+    -------
+    gui : instance of VolSourceEstimateViewer
+        The graphical user interface (GUI) window.
+    """
+    import numpy as np
+    from ..viz.backends._utils import _qt_app_exec
+    from ._stc import VolSourceEstimateViewer
+    from qtpy.QtWidgets import QApplication
+    # get application
+    app = QApplication.instance()
+    if app is None:
+        app = QApplication(['Source Estimate Viewer'])
+
+    # cast to integers to lower memory usage, use custom complex data
+    # type if necessary
+    complex_dtype = np.dtype([('re', np.int64), ('im', np.int64)])
+    data = list()
+    for inner_stcs in (stcs if np.iterable(stcs) else [stcs]):
+        inner_data = list()
+        for stc in (inner_stcs if np.iterable(inner_stcs) else [inner_stcs]):
+            if np.iscomplexobj(stc.data):
+                stc_data = np.zeros(stc.data.shape, complex_dtype)
+                stc_data['re'] = stc.data.real * 1e16
+                stc_data['im'] = stc.data.imag * 1e16
```

Comment:
where is this 1e16 coming from?

Reply:
If you cast to integers without rescaling, you mostly get zeros so this is just so that you can see your data but use integers for speed like is often done for MRIs.